<launch>

<arg name="scan_topic" default="/img_scan"/>
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
    <param name="base_frame" value="base_link"/>
    <param name="odom_frame" value="/odom"/>
    <param name="map_frame" value="map"/>
    <param name="linearUpdate" value="0.0"/>
    <param name="angularUpdate" value="0.0"/>
    <param name="map_update_interval" value="0.01"/>
    <param name="particles" value="100"/>
    
    <param name="maxUrange" value="10"/>
    <param name="sigma" value="0.05"/>
    <param name="kernelSize" value="0.1"/>
    <param name="lstep" value="0.5"/>
    <param name="astep" value="0.5"/>
    <param name="iterations" value="1"/>
    <param name="lsigma" value="0.075"/>
    <param name="ogain" value="3.0"/>
    <param name="lskip" value="0"/>
    <param name="minimumScore" value="100"/>
    <param name="srr" value="0.1"/>
    <param name="srt" value="0.2"/>
    <param name="str" value="0.1"/>
    <param name="stt" value="0.2"/>
    
    <param name="temporalUpdate" value="0.05"/>
    <param name="resampleThreshold" value="0.05"/>
    
    <!-- <param name="xmin" value="-50.0"/>
    <param name="ymin" value="-50.0"/>
    <param name="xmax" value="50.0"/>
    <param name="ymax" value="50.0"/>
    <param name="delta" value="0.01"/>
    <param name="llsamplerange" value="0.01"/>
    <param name="llsamplestep" value="0.01"/>
    <param name="lasamplerange" value="0.005"/>
    <param name="lasamplestep" value="0.005"/> -->
    <remap from="scan" to="$(arg scan_topic)"/>
    <remap from="map" to="/my_robot/map"/>
    </node>

<node pkg="rviz" type="rviz" name="rviz" args="-d $(find lane_mapping)/cfg/rvizconf.rviz"/>
<node pkg="lane_mapping" type="laser_broadcaster" name="laser_broadcaster"/>

</launch>